optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -689,9 +647,17 @@ def tune_params(self, target_param_names, n_trials, sampler, step_name):
             if t.state in (optuna.trial.TrialState.COMPLETE, optuna.trial.TrialState.PRUNED)
         ]
         _n_trials = n_trials - len(complete_trials)
+
+        if self.start_time is None:
+            self.start_time = time.time()
```

Comment:
[question] Is there any reason to initialize `self.start_time` in `tune_params()`?
I think it's simpler to initialize it in `__init__()`.

Reply:
Thank you for your question.
I think users may have a gap of `time_budget`. For example, users can add time-consuming computation between `__init__` and `run()`. In that case, the actual time limitation will be shorter than the value of `time_budget`. So, I initialize `self.start_time` right before the `study.optimize`.

But it makes implementation complicated, and I think we can ignore the gap.

Reply:
Thanks for your explanation. It makes sense. Looks good to me on this part.